src/ApplicationInsights.Kubernetes/ContainerIdProviders/EnvironmentVariableContainerIdProvider.cs

```diff
@@ -0,0 +1,24 @@
+#nullable enable
+
+using System;
+using Microsoft.ApplicationInsights.Kubernetes.Debugging;
+
+namespace Microsoft.ApplicationInsights.Kubernetes.ContainerIdProviders
+{
+    internal class EnvironmentVariableContainerIdProvider : IContainerIdProvider
+    {
+        private readonly ApplicationInsightsKubernetesDiagnosticSource _logger = ApplicationInsightsKubernetesDiagnosticSource.Instance;
+
+        private const string EnvironmentVariableName = "ContainerId";
```

Comment:
Consider making this a construction parameter, with `ContainerId` being a default